src/optimistic-oracle/optimistic-oracle.spec.js

```diff
@@ -0,0 +1,389 @@
+
+// "BAD" identifier encoded as a Bytes32 value
+const BAD_IDENTIFIER_DATA = '0x4241440000000000000000000000000000000000000000000000000000000000';
+
+const MOCK_PRICE = '99999999999999999999';
+
+const optimisticOracleAbi = getAbi('OptimisticOracle');
+
+// create interface
+const iface = new ethers.utils.Interface(optimisticOracleAbi);
+
+// mock the getPrice() function from optimistic-oracle.js
+async function mockGetPrice(identifier) {
+  if (identifier === BTC_IDENTIFIER_STRING) {
+    return MOCK_PRICE;
+  }
+
+  throw new Error(`Unknown identifier ${identifier}`);
+}
+
+async function mockGetPriceBadResponse(identifier) {
+  throw new Error(`Timeout or bad response to price feed request for identifier ${identifier}`);
+}
+
+describe('UMA optimistic oracle validation agent', () => {
+  let handleTransaction;
+  let optimisticOracleAddress = null;
+
+  it('should create price feeds for supported assets w/o throwing any exceptions', async () => {
+    // supported list of assets
+    const idList = [
+      'ETH/BTC',
+      'COMP/USD',
+      'COMPUSD',
+      'USDETH',
+      'ETHUSD',
+      'USDBTC',
+      'BTCUSD',
+      'USDPERL',
+      'BCHNBTC',
+      'STABLESPREAD',
+      'STABLESPREAD/USDC',
+      'STABLESPREAD/BTC',
+      'ELASTIC_STABLESPREAD/USDC',
+      'GASETH-TWAP-1Mx1M',
+      'GASETH-FEB21',
+      'GASETH-MAR21',
+      'COMPUSDC-APR-MAR28/USDC',
+      'BTCDOM',
+      'ALTDOM',
+      'AMPLUSD',
+      'DEFI_PULSE_TOTAL_TVL',
+      'DEFI_PULSE_SUSHI_TVL',
+      'DEFI_PULSE_UNISWAP_TVL',
+      'SUSHIUNI',
+      'CNYUSD',
+      'EURUSD',
+      'PHPDAI',
+      'ETH-BASIS-6M/USDC',
+      'ETH-BASIS-3M/USDC',
+      'BTC-BASIS-6M/USDC',
+      'BTC-BASIS-3M/USDC',
+      'USD/bBadger',
+      'USD-[bwBTC/ETH SLP]',
+      'XAUPERL',
+      'XAUUSD',
+      'uSTONKS_APR21',
+      'DIGGBTC',
+      'DIGGETH',
+      'DIGGUSD',
+      'USDAAVE',
+      'AAVEUSD',
+      'USDLINK',
+      'LINKUSD',
+      'USDSNX',
+      'SNXUSD',
+      'USDUMA',
+      'UMAUSD',
+      'USDUNI',
+      'UNIUSD',
+      'USDOCEAN',
+      'OCEANUSD',
+      'USDBTC_18DEC',
+      'STABLESPREAD/USDC_18DEC',
+      'BCHNBTC_18DEC',
+      'ETHBTC_FR',
+      'BALUSD',
+      'XSUSHIUSD',
+      'uSTONKS_JUN21',
+      'PUNKETH_TWAP',
+      'USDXIO',
+      'iFARMUSD',
+      'USDiFARM',
+      'USDDEXTF',
+      'DEXTFUSD',
+      'uSTONKS_0921',
+      // ibBTC currently not supported - requires NODE_URL_137 env variable to be set
+      // 'ibBTC/BTC',
+      // 'BTC/ibBTC',
+      // 'ibBTC/USD',
+      // 'USD/ibBTC',
+      'GASETH-0921',
+    ];
+
+    /* eslint-disable no-await-in-loop */
+    for (let i = 0; i < idList.length; i++) {
+      const identifier = idList[i];
+      const args = {
+        identifier,
+        config: {
+          cryptowatchApiKey: CRYPTOWATCH_API_KEY,
+          defipulseApiKey: DEFIPULSE_API_KEY,
+          tradermadeApiKey: TRADERMADE_API_KEY,
+          cmcApiKey: CMC_API_KEY,
```

Comment:
I find this so curious. seems like a lot of extra work to keep the same names ;) no change suggestion, I am only contemplating